### ISSUE TITLE
Epic: Self healing on startup

### ISSUE THREAD

**Issue by dmacvicar** [codenotary/immudb#823]
If for some reason (unexpected shutdown and/or not running in sync mode) the index and/or the hash tree need to be regenerated, immudb should detect this situation at startup and correct it, by regenerating everything before starting up.

Output is important, so this recovery mode should tell this is happening.

Other cards which can be split out from this:

* While optimizations are important, the first step is to recover, so finding out where to truncate the index and/or trees to start over is a nice to have.
* For monitoring purposes, it may be interesting to expose if the server is running on recovery or not, as a metric. ([example from other software](https://docs.rudderstack.com/user-guides/administrators-guide/monitoring-and-metrics#recovery-mode))
